controllers/upgrade_handlers.go

```diff
@@ -208,9 +209,19 @@ func (r *ImageBasedUpgradeReconciler) postPivot(ctx context.Context, ibu *lcav1a
 	return doNotRequeue(), nil
 }
 
+// getSortedBackupsFromConfigmap wrapper to call the implementation in BackupRestore
+var getSortedBackupsFromConfigmap = func(r *ImageBasedUpgradeReconciler, ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) ([][]*velerov1.Backup, error) {
+	return r.BackupRestore.GetSortedBackupsFromConfigmap(ctx, ibu.Spec.OADPContent)
+}
+
+// startOrTrackBackup wrapper to call the implementation in BackupRestore
+var startOrTrackBackup = func(r *ImageBasedUpgradeReconciler, ctx context.Context, backups []*velerov1.Backup) (*backuprestore.BackupTracker, error) {
+	return r.BackupRestore.StartOrTrackBackup(ctx, backups)
+}
+
 // handleBackup manages backup flow and returns with possible requeue
 func (r *ImageBasedUpgradeReconciler) handleBackup(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
-	sortedBackupGroups, err := r.BackupRestore.GetSortedBackupsFromConfigmap(ctx, ibu.Spec.OADPContent)
```

Comment:
We have mockgen module. You can create an interface for BackupRestore and let mockgen generate the mocks. see https://github.com/openshift-kni/lifecycle-agent/tree/main/internal/ostreeclient

Reply:
Ah yeah totally forgot about that. Was trying to avoid imports but yeah makes sense to use that! Will update